src/gui/tray/Window.qml

```diff
@@ -39,6 +39,15 @@ Window {
         folderStateIndicator.source = UserModel.isUserConnected(UserModel.currentUserId)
                 ? Style.stateOnlineImageSource
                 : Style.stateOfflineImageSource
+        
+        currentAccountStatusIndicator.source = ""
+        currentUserStatus.visible = UserModel.isUserConnected(UserModel.currentUserId) &&
+                UserModel.currentUser.serverHasUserStatus
+        currentAccountStatusIndicatorMouseHover.visible = currentUserStatus.visible
+        currentAccountStatusIndicatorBackground.visible = currentUserStatus.visible

```

Comment:
Something is fishy in that code, those visible properties have property bindings below, but here and in the next hunk you overwrite them. It's better if you use property bindings all the way. Also I suspect this is doable here.

Reply:
in #3045 Alex reported: "right after connecting to my account, the "Online" status is not displayed in the header, I always have to close and re-open the tray window to get this default "Online" status displayed".
I thought handling  those elements in onVisibleChanged/onRefreshCurrentUserGui would fix this behavior and it did.

Reply:
@camilasan I'd still double-check the consequences. As far as I remember - the moment a value is assigned to a variable - bindings to that variable get destroyed. Will the further change of data in the model update the variables then? After any of these events (onVisibleChanged, onRefreshCurrentUserGui) are fired, I guess, bindings are becoming dead?